crates/polars-core/src/frame/mod.rs

```diff
@@ -61,6 +61,38 @@ pub enum UniqueKeepStrategy {
     Any,
 }
 
+fn ensure_names_unique<T, F>(items: &[T], mut get_name: F) -> PolarsResult<()>
+where
+    F: FnMut(&T) -> &str,
+{
+    // Always unique.
+    if items.len() <= 1 {
+        return Ok(());
+    }
+
+    // Too small to be worth spawning a hashmap for, this is at most 6 comparisons.
+    if items.len() <= 4 {
+        for i in 0..items.len() - 1 {
+            let name = get_name(&items[i]);
+            for other in items.iter().skip(i + 1) {
+                if name == get_name(other) {
+                    polars_bail!(duplicate = name);
+                }
+            }
+        }
+    }
+
+    let mut names = PlHashSet::with_capacity(items.len());
+    for item in items {
+        let name = get_name(item);
+        if !names.insert(name) {
+            polars_bail!(duplicate = name);
+        }
+    }
```

Comment:
I might be missing something obvious here but shouldn't this be in an `else` statement? Or return early in the `if` statement above?

Reply:
`polars_bail!` *is* an early return. And `HashSet::insert` returns `false` if the value was already contained in the set. So unless I'm missing something, it is correct.

Reply:
Oh I see now what you mean, yes it's missing a `return Ok(());` in the earelier if `len <= 4` case, good catch!

Reply:
Should be fixed now.